### PR TITLE
[Feature][Doris] Support multi-table source read

### LINKED ISSUE #5731: [Feature][Doris] Support multi-table source read
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

[Doris] Support multi-table source read

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by happyboy1024**
### Purpose of this pull request

Support Doris multi read and close #5731
### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/connector-v2/source/Doris.md
Comment (hailin0) [inline docs/en/connector-v2/source/Doris.md:19]: update cn docs

Comment (happyboy1024) [reply docs/en/connector-v2/source/Doris.md:19]: > update cn docs

done.

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/reader/DorisSourceReader.java
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/reader/DorisSourceReader.java]: I think we should throw exception.

**Review by Hisoka-X — COMMENTED** [review]
LGTM except one minior problem. Thanks @happyboy1024 !

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/reader/DorisSourceReader.java
Comment (happyboy1024) [reply seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/reader/DorisSourceReader.java]: > I think we should throw exception.
Do we need to terminate directly here? Or do we add the warn log output, what do you think?

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/reader/DorisSourceReader.java]: An exception should be thrown because if there is no bug with the connector, it should never be null

Comment (happyboy1024) [reply seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/source/reader/DorisSourceReader.java]: > An exception should be thrown because if there is no bug with the connector, it should never be null

Ok, I see what you mean. I'll adjust. thx

**Review by Hisoka-X — APPROVED** [review]
LGTM if ci passes. Thanks @happyboy1024 !

**Comment by hailin0** [comment]
waiting for ci passed

**Review by hailin0 — APPROVED** [review]

**Pull request merged**
